.travis.yml

```diff
@@ -1,6 +1,36 @@
+sudo: false
 language: cpp
-script: autoreconf -i && ./configure && make check && ./test-program
+compiler:
+  - gcc
+  - clang
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      - gcc-4.9
+      - g++-4.9
+      - clang
+before_install:
+  - if [ "$CXX" = "g++" ]; then 
+      export CXX="g++-4.9" CC="gcc-4.9";
+      export CXXFLAGS="-std=c++11";
+    elif [ "$CXX" = "clang++" ]; then
+      export CXXFLAGS="-std=c++11 -stdlib=libc++ -DGTEST_USE_OWN_TR1_TUPLE=1";
+    fi
```

Comment:
can we move those into the Makefile and the gtest build utils script? Or is this a specific travis problem? I'd much rather have it close to the actual build if this will be a problem outside of Travis.